app/assets/javascripts/state_change_reasons.js

```diff
@@ -1,23 +1,15 @@
-// Global SCAPE.message method
+
 (function($, window, undefined){
   "use strict";
 
-  var displayReason = function() {
-    var new_state = this.value;
-    if($('.reason:visible').length === 0) {
-      $('#'+new_state+'_reasons').slideDown('slow').find('select:disabled').prop('disabled',false);
-    }
-    else {
-      $('.reason').not('#'+new_state+'_reasons').slideUp('slow', function(){
-        $('#'+new_state+'_reasons').slideDown('slow').find('select:disabled').prop('disabled',false);
-      }).prop('disabled',true);
-    }
-  };
-
-  $(document).ready(function(){
-    $('#state-changer').find('#state').on('change', displayReason);
-  });
-
+  var updateDisplay = function(new_state, speed) {
+    $('.reason').not('#' + new_state + '_reasons').slideUp(speed)
+    $('#' + new_state + '_reasons').slideDown(speed)
+  }
 
```

Comment:
Simplified by removing need to disable form-fields. Also reduces the risk of accidentally submitting the wrong information.